### DISCUSSION TITLE
Can you add ability to turn on user feedback?

### DISCUSSION THREAD

**Discussion by juslintek** [stayallive/wp-sentry#110] [Ideas]
Hi, can you add code snippet with switch to enable user feedback?

**Comment by Brayden967** [comment]
This would be a great feature. There is a lot of documentation on Sentry for adding user feedback, I tried using the Browser JS SDK but it did not work, this will have to be done with some sort of PHP integration.

**Comment by stayallive** [comment]
Something like this on your error page should work.

```php
<?php $sentry = class_exists('WP_Sentry_Php_Tracker') 
    ? WP_Sentry_Php_Tracker::get_instance()->get_client() 
	: null;

if ($sentry !== null && !empty($sentry->getLastEventID())) : ?>
    <p>Error ID: <?php echo $sentry->getLastEventID(); ?></p>

    <!-- Sentry JS SDK 2.1.+ required -->
    <script src="https://cdn.ravenjs.com/3.26.2/raven.min.js" crossorigin="anonymous"></script>

    <script>
        Raven.showReportDialog({
            eventId: '<?php echo $sentry->getLastEventID(); ?>',
            // use the public DSN (dont include your secret!)
            dsn: '<?php echo WP_SENTRY_PUBLIC_DSN; ?>',
            user: {
                'name': 'Jane Doe',
                'email': 'jane.doe@example.com',
            }
        });
    </script>
<?php endif; ?>
```

If you are tracking JS errors including the Raven SDK might not be needed depending on your error template.

**Comment by Brayden967** [comment]
This is perfect, once a PHP error is thrown (not fatal), the pop-up box does appear. I tested putting this code in my themes header and footer and both work. 

One suggestion would be too replace your user variables with this to allow more accurate reporting and follow-up.

*I created a pull request with this added to the ReadMe for future reference

```php
global $current_user;
$current_user = wp_get_current_user();
 user: { 
                'name': '<?php echo $current_user->user_login; ?>',
                'email': '<?php echo $current_user->$current_user->user_email; ?>',
            }

**Comment by andreydruz** [comment]
Hello,

It is possible to add user feedback on version 3.x? 
Same as here https://github.com/stayallive/wp-sentry/issues/21#issuecomment-402049699 ?
How to set error page? Redirect in .htaccess to some php page? If so, it may be better to add this page in plugin and add it to readme.
